File: backend/shoppro-core/src/main/java/com/shoppro/service/MarketingService.java
package com.shoppro.service;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.shoppro.entity.MarketingTask;

/**
 * 营销服务接口
 */
public interface MarketingService {
    /**
     * 创建营销任务
     */
    MarketingTask createTask(MarketingTask task);

    /**
     * 启动/调度任务
     */
    void startTask(Long taskId);

    /**
     * 暂停任务
     */
    void stopTask(Long taskId);

    /**
     * 分页查询任务
     */
    Page<MarketingTask> pageTasks(int pageNo, int pageSize, String name);

    /**
     * 获取发送记录 (此处简化为获取任务摘要)
     */
    MarketingTask getTaskSummary(Long taskId);
}

File: shoppro-app/src/services/marketingService.ts
import { http } from './http';

export interface MarketingTask {
    id?: number;
    name: string;
    type: 'email' | 'sms' | 'notification';
    status?: 'draft' | 'scheduled' | 'running' | 'completed' | 'failed';
    targetAudience?: string; // JSON string or description
    content?: string;
    scheduleTime?: string;
    createdAt?: string;
    updatedAt?: string;
}

export const marketingService = {
    /**
     * Create a new marketing task
     */
    createTask: (task: MarketingTask) => {
        return http.post<MarketingTask>('/marketing/tasks', task);
    },

    /**
     * Start task scheduling
     */
    startTask: (id: number) => {
        return http.post<void>(`/marketing/tasks/${id}/start`);
    },

    /**
     * Stop task scheduling
     */
    stopTask: (id: number) => {
        return http.post<void>(`/marketing/tasks/${id}/stop`);
    },

    /**
     * Page query marketing tasks
     */
    getTasks: (params: { pageNo?: number; pageSize?: number; name?: string }) => {
        return http.get<{ records: MarketingTask[]; total: number; current: number }>('/marketing/tasks', { params });
    },

    /**
     * Get task execution summary
     */
    getTaskSummary: (id: number) => {
        return http.get<MarketingTask>(`/marketing/tasks/${id}/summary`);
    }
};

export default marketingService;
